plugin/evm/vm.go

```diff
@@ -742,3 +751,49 @@ func (vm *VM) readLastAccepted() (common.Hash, error) {
 		return lastAcceptedHash, nil
 	}
 }
+
+// handleUpgradeBytes applies the configuration of upgradeBytes
+// to the chain config. If specified upgradeBytes are not compatible
+// with previously used upgradeBytes, an error is returned.
+// If they are compatible, they are persisted and should be specified
+// again or replaced with a new, compatible configuration.
+func (vm *VM) handleUpgradeBytes(upgradeBytes []byte) error {
```

Comment:
Can we instead pass this into `core.BlockChain` to be handled in the constructor in the same place that we check if the chain config is valid?

I think it makes sense to. keep these two checks in the same place.

Reply:
This would mean we can add the `upgradeBytesKey` to the DB schema and we can simply unmarshal the upgrade bytes into the `UpgradeConfig` and pass the whole `ChainConfig` into the `*core.BlockChain` object when it gets constructed the same way we do now.